controllers/healthcheck/machine.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthcheck
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate counterfeiter . Client
+type Client interface {
+	Delete(context.Context, runtime.Object, ...client.DeleteOption) error
+}
+
+func NewUnhealthyMachineDeletionRemediator(client Client, logger logr.Logger) *UnhealthyMachineDeletionRemediator {
+	return &UnhealthyMachineDeletionRemediator{
+		client: client,
+		logger: logger,
+	}
+}
+
+type UnhealthyMachineDeletionRemediator struct {
```

Comment:
```suggestion
type OwnerControllerStrategy struct {
```

If we have these in a package called `remediators` then in the future we could have multiple strategies in there

Reply:
This is still the deletion strategy though, right? They will all be owner controller strategies in that we expect the owner controller to be the one running the remediator.

Reply:
For now, yes that's correct. In the future if we have multiple strategies I can see us having some logic to pick a strategy based on what the MHC is configured to. One of them could be a no-op strategy, which falls in line with external remediation.

We can always rework it later, but if we structure the code to be composable today, it's a good investment for future iterations, that's also why I was suggesting to have a `remediation.Strategy` interface above.

Reply:
The package can/should be under `controllers/healthcheck/remediation` potentially